### PR TITLE
cleanup provider digests

### PR DESCRIPTION
**Pull request by slontis**
Added some missing #ifdef NO_XXX around some of the digest functions.
Renamed core_mkdigest.h to digestcommon.h
Added ERR_raise() to set/get params for digest.
Moved common code for get_params/gettable_params into digest_common.c
Renamed #defines in digestcommon.
Removed null_prov.c (It should not be needed)

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: providers/common/include/internal/digestcommon.h [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+const OSSL_PARAM *digest_default_gettable_params(void);
+int digest_default_get_params(OSSL_PARAM params[], int blksz, int paramsz,
+                              unsigned long flags);
+
+# ifdef __cplusplus
+}
+# endif
+
+#endif /* OSSL_DIGESTCOMMON_H */
```
Comment (levitte) [inline providers/common/include/internal/digestcommon.h]: Regarding names, I wonder if `OSSL_FUNC_*` and `OSSL_DISPATCH_*` should be renamed to something a little more provider specific. After all, these are macros we don't publish anywhere...

Comment (slontis) [reply providers/common/include/internal/digestcommon.h]: renamed and collapsed some macros that were not shared.
Also updated the docs.

File: providers/default/digests/blake2_prov.c [hunk 2]
```diff
@@ -7,10 +7,15 @@
...
 
-#include <openssl/crypto.h>
-#include "internal/blake2.h"
-#include "internal/core_mkdigest.h"
-#include "internal/provider_algs.h"
+#include <openssl/opensslconf.h>
+
+#ifdef OPENSSL_NO_BLAKE2
+NON_EMPTY_TRANSLATION_UNIT
+#else
```
Comment (levitte) [inline providers/default/digests/blake2_prov.c]: Why?  If you look in `providers/default/digests/build.info`, you'll see that disabled algos aren't built (if they are, there's a bug in build.info processing)

This goes for all places where you added this.

File: providers/legacy/digests/md2_prov.c [hunk 3]
```diff
@@ -6,13 +6,18 @@
  * in the file LICENSE in the source distribution or at
  * https://www.openssl.org/source/license.html
  */
+#include <openssl/opensslconf.h>
 
-#include <openssl/crypto.h>
-#include <openssl/md2.h>
+#ifdef OPENSSL_NO_MD2
+NON_EMPTY_TRANSLATION_UNIT
+#else
```
Comment (levitte) [inline providers/legacy/digests/md2_prov.c]: The same goes for legacy

File: providers/default/digests/blake2_prov.c (hunk 2 above)
Comment (slontis) [reply providers/default/digests/blake2_prov.c]: hangs head in shame - and goes and reverts changes...

**Comment by slontis** [comment]
Thanks Richard. Merged to master.

**Pull request closed without merging**